contracts/BSCValidatorSet.sol

```diff
@@ -360,56 +388,100 @@ contract BSCValidatorSet is IBSCValidatorSet, System, IParamSubscriber, IApplica
     return CODE_OK;
   }
 
-  function shuffle(address[] memory validators, uint256 epochNumber, uint startIdx, uint offset, uint limit, uint modNumber) internal pure {
-    for (uint i = 0; i<limit; i++) {
-      uint random = uint(keccak256(abi.encodePacked(epochNumber, startIdx+i))) % modNumber;
-      if ( (startIdx+i) != (offset+random) ) {
-        address tmp = validators[startIdx+i];
-        validators[startIdx+i] = validators[offset+random];
-        validators[offset+random] = tmp;
+  function shuffle(address[] memory validators, bytes[] memory voteAddrs, uint256 epochNumber, uint startIdx, uint offset, uint limit, uint modNumber) internal pure {
+    for (uint i = 0; i < limit; i++) {
+      uint random = uint(keccak256(abi.encodePacked(epochNumber, startIdx + i))) % modNumber;
+      if ((startIdx + i) != (offset + random)) {
+        address tmpAddr = validators[startIdx + i];
+        bytes memory tmpBLS = voteAddrs[startIdx + i];
+        validators[startIdx + i] = validators[offset + random];
+        validators[offset + random] = tmpAddr;
+        voteAddrs[startIdx + i] = voteAddrs[offset + random];
+        voteAddrs[offset + random] = tmpBLS;
+      }
+    }
+  }
+
+  function getLivingValidators() external initValidatorExtraSet override returns (address[] memory, bytes[] memory) {
```

Comment:
this is view func, we could not remove view

Reply:
Fixed